compiler/lib/src/main/scala/analysis/CheckSemantics/CheckUses.scala

```diff
@@ -133,7 +148,7 @@ object CheckUses extends UseAnalyzer {
         for {
           _ <- {
             Result.annotateResult(
-              constantUse(a, itu.asExprNode, itu.name),
```

Comment:
This looks like it introduces a regression. In the old code, `constantUse` would fail if the definition referred to did not exist. Now it looks like that check is gone, because the `constantUse` analysis here is more permissive. I wonder if the `BasicUseAnalyzer` should have two kinds of `constantUse` functions: one that works the way it worked before, where we are expecting a constant use and we fail if it's not there; and one that doesn't a priori know whether a dot expression is a constant use, so allows it not to be. The second one could try the first one and check for value selection if it fails. Here it seems we should be using the first one.

Reply:
After discussion, we'll keep the existing code structure. Here we will (1) call `constantUse` and (2) check that the qualified identifier really is a constant use, i.e., it is represented in the use-def map.